uxarray/grid.py

```diff
@@ -0,0 +1,279 @@
+"""uxarray grid module."""
+import os
+import xarray as xr
+import numpy as np
+from pathlib import PurePath
+
+
+class Grid:
+    """A class for uxarray grid object.
+
+    Examples
+    ========
+    import uxarray as ux
+    # open an exodus file with Grid object
+    mesh = ux.Grid("filename.g")
+
+    # save as ugrid file
+    mesh.saveas("outfile.ug")
+    """
+
+    # Import read/write methods from files in this folder
+    from ._exodus import read_exodus, write_exodus
+    from ._ugrid import read_ugrid, write_ugrid
+    from ._shpfile import read_shpfile
+    from ._scrip import read_scrip
+
+    def __init__(self, *args, **kwargs):
+        """Initialize grid variables, decide if loading happens via file, verts
+        or gridspec If loading from file, initialization happens via the
+        specified file.
+        Args: input file name with extension as a string
+              vertex coordinates that form one face.
+        kwargs: can be a dict initializing specific variables: kwargs = {"concave" : True, "islatlon" : True"}
+
+        Raises:
+            RuntimeError: File not found
+        """
+        # initialize possible variables
+        for key, value in kwargs.items():
+            if key == "latlon":
+                self.islatlon = value
+            elif key == "gridspec":
+                self.gridspec = value
+
+        self.filepath = None
+        self.gridspec = None
+        self.vertices = None
+        self.islatlon = None
+        self.concave = None
+        self.meshFileType = None
+
+        # this xarray variable holds an existing external netcdf file with loaded with xarray
+        self.ext_ds = None
+
+        # internal uxarray representation of mesh stored in internal object in_ds
+        self.in_ds = xr.Dataset()
+
+        # determine initialization type - string signifies a file, numpy array signifies a list of verts
+        in_type = type(args[0])
+
+        # check if initializing from verts:
```

Comment:
My personal thought is that following how this code determines if to initialize from verts, gridspec, etc. is not very easy, i.e. the code does not seem easily readable. We may consider simplifying it (maybe my recommendation of using only kwags helps here) as well as putting more explanation comments either here or into the function doc strings.